### PR TITLE
Add pt-BR translation

### PR DESCRIPTION
**Pull request by charbelrami**
Translation into Brazilian Portuguese.
:smile:

### CODE REVIEW

**Comment by forresst** [comment]
Awesome! Do you know anyone that could review it? I will review that the format is correct. (Not the Brazilian Portuguese of course, i'm french).

File: pt_BR/readme.md [hunk 1]
```diff
@@ -0,0 +1,750 @@
...
+test('bar', async t => {
+	t.plan(1);
+
+	const bar = Promise.resolve('bar').then(delay(200));
+
+	t.is(await bar, 'bar');
+});
+```
+
+<img src="screenshot.png" width="150" align="right">
```
Comment (forresst) [inline pt_BR/readme.md]: Bad link, replace : 
``
By
``

File: pt_BR/readme.md [hunk 2]
```diff
@@ -0,0 +1,750 @@
...
+    "require": ["babel-core/register", "coffee-script/register"]
+  }
+}
+```
+
+Argumentos passados para a CLI sempre terão precedência sobre a configuração em `package.json`.
+
+## Documentação
+
+Testes são executados de forma assíncrona e exigem que você retorne um objeto assíncrono suportado (uma promessa, ou um [observável](https://github.com/zenparsing/zen-observable)). Nós *altamente* recomendamos o uso de [async functions](#async-function-support); Elas produzem código assíncrono conciso e legível, e retornam, implicitamente, uma promessa, assim você não precisa fazê-lo.
```
Comment (forresst) [inline pt_BR/readme.md]: Bad link, replace 
(#async-function-support)
By
(#suporte-para-funções-assíncronas)

File: pt_BR/readme.md [hunk 3]
```diff
@@ -0,0 +1,750 @@
...
+
+Se você é incapaz de usar promessas ou outros objetos assíncronos suportados, você pode ativar o "callback mode", definindo seu teste com `test.cb([title], fn)`. Testes declarados desta forma **devem** deve ser manualmente encerrados com `t.end()`. Este modo destina-se principalmente para testar APIs do estilo callback.
+
+Você deve definir todos os testes de forma síncrona. Eles não podem ser definidos dentro de `setTimeout`, `setImmediate`, etc.
+
+Arquivos de teste são executados a partir de seu diretório atual, portanto, [`process.cwd()`](https://nodejs.org/api/process.html#process_process_cwd) sempre é o mesmo que [`__dirname`](https://nodejs.org/api/globals.html#globals_dirname). Você pode apenas usar caminhos relativos ao invés de fazer `path.join(__dirname, 'relative/path')`.
+
+### Teste de anatomia
+
+Para criar um teste, você chama a função `test` após o `require` do AVA e passa um nome de teste opcional e uma função que contém a execução do teste. A função fornecida recebe o contexto como primeiro argumento, onde você pode chamar os distintos métodos de AVA [assertions](#assertions).
```
Comment (forresst) [inline pt_BR/readme.md]: Bad link, replace
(#assertions)
by
(#asserções)

File: pt_BR/readme.md [hunk 4]
```diff
@@ -0,0 +1,750 @@
...
+	}, 100);
+});
+```
+
+#### ATENÇÃO: Ruptura recente.
+
+AVA não suporta mais encerrar automaticamente testes via `t.plan(...)`. Isso ajuda a evitar falsos positivos se você adiciona asserções, mas se esquece de aumentar sua contagem de plano.
+
+```js
+// This no longer works
```
Comment (forresst) [inline pt_BR/readme.md]: Comments can be translated

File: pt_BR/readme.md [hunk 5]
```diff
@@ -0,0 +1,750 @@
...
+
+test('auto ending is dangerous', t => {
+	t.plan(2);
+
+	t.pass();
+	t.pass();
+
+	// auto-ending after reaching the planned two assertions will miss this final one
```
Comment (forresst) [inline pt_BR/readme.md]: Comments can be translated

File: pt_BR/readme.md [hunk 6]
```diff
@@ -0,0 +1,750 @@
...
+
+```js
+test.cb('explicitly end your tests', t => {
+	t.plan(2);
+
+	t.pass();
+	t.pass();
+
+	setTimeout(() => {
+		// This failure is now reliably caught.
```
Comment (forresst) [inline pt_BR/readme.md]: Comments can be translated

File: pt_BR/readme.md [hunk 7]
```diff
@@ -0,0 +1,750 @@
...
+});
+```
+
+### Hooks before e after
+
+Quando instalação e/ou subdivisão é necessária, você pode usar `test.before()` e `test.after()`, da mesma maneira que `test()`. A função de teste fornecida para `test.before()` e `test.after()` é chamada antes/após todos os testes. Você também pode usar `test.beforeEach()` e `test.afterEach()` se você precisa de configuração/subdivisão para cada teste. Hooks são executados serialmente no arquivo teste. Adicione quantos quiser. Opcionalmente, você pode especificar um título que é mostrado em caso de falha.
+
+```js
+test.before(t => {
+	// this runs before all tests
```
Comment (forresst) [inline pt_BR/readme.md]: Comments can be translated

File: pt_BR/readme.md [hunk 8]
```diff
@@ -0,0 +1,750 @@
...
+});
+
+test.before(t => {
+	// this runs after the above, but before tests
```
Comment (forresst) [inline pt_BR/readme.md]: Comments can be translated

File: pt_BR/readme.md [hunk 9]
```diff
@@ -0,0 +1,750 @@
...
+});
+
+test.after('cleanup', t => {
+	// this runs after all tests
```
Comment (forresst) [inline pt_BR/readme.md]: Comments can be translated

File: pt_BR/readme.md [hunk 10]
```diff
@@ -0,0 +1,750 @@
...
+});
+
+test.beforeEach(t => {
+	// this runs before each test
```
Comment (forresst) [inline pt_BR/readme.md]: Comments can be translated

File: pt_BR/readme.md [hunk 11]
```diff
@@ -0,0 +1,750 @@
...
+});
+
+test.afterEach(t => {
+	// this runs after each test
```
Comment (forresst) [inline pt_BR/readme.md]: Comments can be translated

File: pt_BR/readme.md [hunk 12]
```diff
@@ -0,0 +1,750 @@
...
+});
+
+test(t => {
+	// regular test
```
Comment (forresst) [inline pt_BR/readme.md]: Comments can be translated

File: pt_BR/readme.md [hunk 13]
```diff
@@ -0,0 +1,750 @@
...
+#### Transpiling de Módulos Importados
+
+AVA atualmente só faz o transpiling de testes que você pede para serem executados. *Não fará o transpiling de módulos que você importa (```import```) de fora do teste.*  Embora existam razões válidas para esta abordagem, pode não ser o que você espera!
+
+Como uma solução simples, você pode usar [Babel's require hook](https://babeljs.io/docs/usage/require/) a fim de fazer transpiling on-the-fly de módulos que são posteriormente importados. Como o AVA oferece suporte para a sintaxe de módulo ES2015, você pode usá-lo para importar o hook requerido em si:
+
+```js
+import test from 'ava';
+import 'babel-core/register';
+import foo from './foo'; // <-- foo can be written in ES2015!
```
Comment (forresst) [inline pt_BR/readme.md]: Comments can be translated

File: pt_BR/readme.md [hunk 14]
```diff
@@ -0,0 +1,750 @@
...
+test(t => {
+	return somePromise().then(result => {
+		t.is(result, 'unicorn');
+	});
+});
+```
+
+### Suporte para função geradora
+
+AVA vem com suporte embutido para [funções geradoras](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/function*).
```
Comment (forresst) [inline pt_BR/readme.md]: Use link Brazilian Portuguese is better (https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Statements/function*)

File: pt_BR/readme.md [hunk 15]
```diff
@@ -0,0 +1,750 @@
...
+
+AVA vem com suporte embutido para [funções assíncronas](https://tc39.github.io/ecmascript-asyncawait/) *(async/await)*.
+
+```js
+test(async function (t) {
+	const value = await promiseFn();
+	t.true(value);
+});
+
+// async arrow function
```
Comment (forresst) [inline pt_BR/readme.md]: Comments can be translated

File: pt_BR/readme.md [hunk 16]
```diff
@@ -0,0 +1,750 @@
...
+ Se você retornar um observável de um teste, AVA automaticamente irá consumí-lo até sua conclusão antes de encerrar o teste.
+
+*Você não precisa usar o "modo callback" ou chamar `t.end()`.*
+
+```js
+test(t => {
+	t.plan(3);
+	return Observable.of(1, 2, 3, 4, 5, 6)
+		.filter(n => {
+			// only even numbers
```
Comment (forresst) [inline pt_BR/readme.md]: Comments can be translated

File: pt_BR/readme.md [hunk 17]
```diff
@@ -0,0 +1,750 @@
...
+});
+```
+
+### Suporte para Callback
+
+AVA suporta o uso de `t.end` como callback final ao usar APIs de callback node-style error-first. AVA irá considerar qualquer valor truthy passado como o primeiro argumento para o `t.end` como sendo um erro. Observe que `t.end` requer "modo callback", que pode ser ativado usando a cadeia `test.cb`.
+
+```js
+test.cb(t => {
+	// t.end automatically checks for error as first argument
```
Comment (forresst) [inline pt_BR/readme.md]: Comments can be translated

File: pt_BR/readme.md [hunk 18]
```diff
@@ -0,0 +1,750 @@
...
+
+### Saída opcional TAP
+
+AVA pode gerar saída TAP através da opção `--tap` para uso com qualquer [TAP reporter](https://github.com/sindresorhus/awesome-tap#reporters).
+
+```
+$ ava --tap | tap-nyan
+```
+
+<img src="media/tap-output.png" width="398">
```
Comment (forresst) [inline pt_BR/readme.md]: Bad link, replace : 
``
By : 
``

File: pt_BR/readme.md [hunk 19]
```diff
@@ -0,0 +1,750 @@
...
+
+
+### Rastreamentos limpos de Pilha
+
+AVA automaticamente remove linhas independentes em rastreamentos de pilha, permitindo que você encontre a fonte de um erro muito mais rápido.
+
+<img src="media/stack-traces.png" width="300">
```
Comment (forresst) [inline pt_BR/readme.md]: Bad link, replace : 
``
By : 
``

File: pt_BR/readme.md [hunk 20]
```diff
@@ -0,0 +1,750 @@
...
+Afirma que `error` é falsy.
+
+## Ignorando asserções
+
+Qualquer asserção pode ser ignorada usando o modificador `skip`. Asserções ignoradas ainda são contadas, e, portanto, não há nenhuma necessidade de mudar a sua contagem de asserção planejada.
+
+```js
+test(t => {
+  t.plan(2);
+  t.skip.is(foo(), 5); // no need to change your plan count when skipping.
```
Comment (forresst) [inline pt_BR/readme.md]: Comments can be translated

File: pt_BR/readme.md [hunk 21]
```diff
@@ -0,0 +1,750 @@
...
+
+Mocha requer a utilização de globais implícitas como `describe` e `it` com a interface padrão (que a maioria das pessoas usam), é muito não-opinativo, inchado, síncrono por padrão, tem execução de teste serial, e lento. Tape e node-tap são muito bons. AVA é altamente inspirado em sua sintaxe. No entanto, ambos executam testes em série e fizeram do [TAP](https://testanything.org) um cidadão de primeira classe que, na minha opinião, tornou as suas bases de código um pouco complicadas e acopladas. A saída do TAP é difícil de ler, então você sempre acaba usando um reporter externo. AVA é altamente opinativo e simultâneo. Vem com um simples reporter padrão e suporta TAP através de um sinalizador CLI.
+
+### Como posso utilizar reporters personalizados?
+
+Use a [`--tap` flag](#optional-tap-output) com qualquer [TAP reporter](https://github.com/sindresorhus/awesome-tap#reporters).
+
+### Como o nome é escrito e pronunciado?
+
+AVA, não Ava nem ava. Pronuncia-se [`/ˈeɪvə/` ay-və](media/pronunciation.m4a?raw=true).
```
Comment (forresst) [inline pt_BR/readme.md]: Bad link, replace : 
`(media/pronunciation.m4a?raw=true)`
By : 
`(https://github.com/sindresorhus/ava/blob/master/media/pronunciation.m4a?raw=true)`

File: pt_BR/readme.md [hunk 22]
```diff
@@ -0,0 +1,750 @@
...
+
+### O que é o plano de fundo do cabeçalho?
+
+[Galáxia de Andrômeda.](https://simple.wikipedia.org/wiki/Andromeda_galaxy)
```
Comment (forresst) [inline pt_BR/readme.md]: This link is better ? (https://pt.wikipedia.org/wiki/Gal%C3%A1xia_de_Andr%C3%B4meda)

**Comment by charbelrami** [comment]
Je vais corriger ces erreurs, merci

**Comment by forresst** [comment]
Except my remarks, it's great work! Thank you (Sorry, I'm not talking a Brazilian word)

**Comment by charbelrami** [comment]
OK, thank you!

**Comment by sindresorhus** [comment]
Awesome work @charbelrami! :)

> Do you know anyone that could review it?

@charbelrami :arrow_up:

**Comment by charbelrami** [comment]
Thank you @sindresorhus :) I'll ask someone to review it.

**Comment by amravazzi** [comment]
1. I would use `Promise` instead of `Promessa`. It is not incorrect, but sounds weird. It would sound the same as translating "IO", "multi-thread" or "callback".
2. You should standardize the gender for `CLI`. In some excerpts, it is written `... NÃO-PADRÃO: ~A~ CLI AVA...` and then right after, `as opções ~do~ CLI`. You should opt for the _male_ gender for "CLI"
3. You should review the ponctuation on this sentence: `Elas produzem código assíncrono conciso e legível, e retornam, implicitamente, uma promessa, assim você não precisa fazê-lo.`
4. On the excerpt `A função fornecida recebe o contexto como primeiro argumento, onde você pode chamar os distintos...`, the word `onde` should be replaced by `no qual`. The word `onde` implies physical places and should only be used within this sort of context.
5. All the misuse of the word `onde` should be reviewed.
6. `... garantir que um número específico de asserções é feito` you should replace the verb `é`on its present tense to its correspondent on the present subjunctive tense `seja`
7. Every non-Portuguese words in the middle of a Portuguese sentence such as `callbacks` or `loops` should be written in _italic_ 
8. Avoid the use of slashes in place of `OU` such as in `test.after() é chamada antes/após todos os`.
9. On the sentence `Você pode usar qualquer módulo de asserção no lugar, ou além, do que vem com o AVA, mas você não será capaz de usar o método .plan(), ainda.`you should review the use of commas.
10. On the excerpt `Lê o seu teste e tenta inferir mais informações a partir do código...`, the verb `Lê` should be preceded by a connector. 
11. On the excerpt `Mesmo que Node.js possa executar IO...` the noun `Node.js` requires a definite article `o`.

**Comment by forresst** [comment]
Sorry, one last point that I forgot, for all files translated  : 

As the [French version](https://github.com/sindresorhus/ava-docs/edit/master/fr_FR/contributing.md) and [Spanish version](https://github.com/sindresorhus/ava-docs/edit/master/es_ES/contributing.md), add a header for the link to the original and to have a link on the differences, for example markdown formatted (Note, I do not speak Brazilian Portuguese, it is a translation in english. It is therefore necessary to translate it into Brazilian Portuguese.) :

``` markdown
___
**Translator's Note**

This is the translation of [contributing.md](https://github.com/sindresorhus/ava/blob/master/contributing.md) file. Here is a [link](https://github.com/sindresorhus/ava/compare/b76f3d513a41edd876f70fc3e61d7d968cd531a6...master#diff-cc4aac3e9be04e0413c9520f223b493c) to differences with the master AVA branch (If clicking the link, no changes were found in the `contributing.md` file, will be that the translation is up to date).
___
```
1. Where the link `contributing.md` points to the file contributing.md from AVA
2. Where the link `link` points to the last commit respective translation (`65ae07c76b2a5927e6bcc00630a691c98f6e7c46`) followed by `...master#diff-` and sha file (`cc4aac3e9be04e0413c9520f223b493c`)

That is to do on the 6 files : code-of-conduct, contributing.md, maintaining.md, readme.md, code-coverage and endpoint-testing.

**Comment by charbelrami** [comment]
OK.
It seems that sha is not required after master in order to compare with the latest version. For example, either https://github.com/sindresorhus/ava/compare/b76f3d513a41edd876f70fc3e61d7d968cd531a6...master#diff-cc4aac3e9be04e0413c9520f223b493c or https://github.com/sindresorhus/ava/compare/b76f3d513a41edd876f70fc3e61d7d968cd531a6...master produces the same result.
Thanks @forresst !

**Comment by forresst** [comment]
@charbelrami 

> It seems that sha is not required after master in order to compare with the latest version. For example, either sindresorhus/ava@b76f3d5...master#diff-cc4aac3e9be04e0413c9520f223b493c or sindresorhus/ava@b76f3d5...master produces the same result.

Interesting ! The behavior has changed on github.
Before, there was a difference, the first link allowed to move themselves on file changes readme.md (if that file had a change), the second has the same behavior as now he move not on the file.

**Comment by forresst** [comment]
@charbelrami 

> > It seems that sha is not required after master in order to compare with the latest version. For example, either sindresorhus/ava@b76f3d5...master#diff-cc4aac3e9be04e0413c9520f223b493c or sindresorhus/ava@b76f3d5...master produces the same result.
> 
> Interesting ! The behavior has changed on github.
> Before, there was a difference, the first link allowed to move themselves on file changes readme.md (if that file had a change), the second has the same behavior as now he move not on the file.

Github has fixed the problem. The behavior is now correct. Thank you to hand in the form `https://github.com/sindresorhus/ava/compare/b76f3d513a41edd876f70fc3e61d7d968cd531a6...master#diff-cc4aac3e9be04e0413c9520f223b493c`

After this modification, it is OK for me.
@sindresorhus Your opinion ?

**Comment by sindresorhus** [comment]
> After this modification, it is OK for me.

@charbelrami Can you do the change :arrow_up:?

And also add a link and your name to the readme of this repo like the other translations?

> Your opinion ?

Looks good :)

**Comment by charbelrami** [comment]
Thanks @forresst @sindresorhus. The translation is up to date :smile:

@forresst, there is no difference between https://github.com/sindresorhus/ava/compare/b76f3d513a41edd876f70fc3e61d7d968cd531a6...master  and https://github.com/sindresorhus/ava/compare/b76f3d513a41edd876f70fc3e61d7d968cd531a6...master#diff-cc4aac3e9be04e0413c9520f223b493c. Maybe I'm missing something :sweat_smile:.  Thank you for clarification.

**Comment by sindresorhus** [comment]
> And also add a link and your name to the readme of this repo like the other translations?

@charbelrami _This_ repo. Fixed it for you ;)

**Comment by sindresorhus** [comment]
@charbelrami Thank you so much! Added you to the repo. Feel free to just commit and push changes directly ;)

**Comment by charbelrami** [comment]
Thank you @sindresorhus!! :)

**Pull request closed without merging**